### ISSUE TITLE
Dhall out-of-date 1.39.0 -> 1.40.1

### ISSUE THREAD

**Issue by toastal** [NixOS/nixpkgs#135519]
###### Checklist

- [x] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name:

current version: 1.39.0
desired version: 1.40.1

###### Notify maintainers

maintainers: @Gabriel439 

###### Note for maintainers

`dhall-toml` and `dhall-csv` too?? <3

**Comment by Gabriel439** [comment]
I'm still new to the Nixpkgs Haskell infrastructure, but I believe this is more of an upstream issue to report against Stackage.  The reason why is that I think the Nixpkgs Haskell package set tries to track Stackage nightly as closely as possible and at the time of this writing [Stackage nightly](https://www.stackage.org/nightly-2021-08-24/cabal.config) is still on `dhall ==1.39.0`

I also haven't been pinged by Stackage to fix any `dhall`-related build failures or bounds failures, so I think the only thing to do here is to just wait until they bump the Dhall version.

**Comment by toastal** [comment]
I don't know how to work with Haskell in Nix either, so I pinged ya. Haskell has always been the most finicky toolchain to me.

**Comment by sternenseemann** [comment]
> I'm still new to the Nixpkgs Haskell infrastructure, but I believe this is more of an upstream issue to report against Stackage. The reason why is that I think the Nixpkgs Haskell package set tries to track Stackage nightly as closely as possible and at the time of this writing [Stackage nightly](https://www.stackage.org/nightly-2021-08-24/cabal.config) is still on `dhall ==1.39.0`

This is correct, however we are currently tracking LTS-18 at the moment (since we don't want to upgrade to GHC 9 yet). Thus we'll upgrade dhall when LTS-18 upgrades dhall (with maybe a week or so of delay).
